web/src/app/ui/kmwuitoolbar.ts

```diff
@@ -1,12 +1,52 @@
 /***
-   KeymanWeb 11.0
-   Copyright 2019 SIL International
+   KeymanWeb 17.0
+   Copyright 2019-2023 SIL International
 ***/
 
+import type { KeymanEngine, KeyboardCookie, UIModule } from 'keyman/app/browser';
+import { type StylesheetManager } from 'keyman/engine/dom-utils';
+
+declare var keyman: KeymanEngine
+
+type MapRegion = {
+  /** The title of the region */
+  t: string,
+
+  /** The geometrical specification, in string form, for the region of the map to highlight */
+  m: string
+}
+
+type ToolbarCookie = {
+  /**
+   * The two-letter code for the currently-selected map region
+   */
+  region: string;
+
+  /**
+   * The number of encoded 'recent'-keyboard entries enumerated within the cookie.
+   */
+  maxrecent: number;
+} & Record<`recent${number}`, string>;
+
+type KeyboardDetail = ReturnType<KeymanEngine['_GetKeyboardDetail']>;
```

Comment:
No points for guessing my favorite recently-learned TS trick.  It came in _really_ helpful for these UI modules.